bson/src/main/org/bson/codecs/OverridableUuidRepresentationCodec.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs;
+
+import org.bson.UuidRepresentation;
+
+/**
+ * A marker interface for {@code Codec} implementations that can derive a new instance of itself that overrides the default
```

Comment:
This might be clearer:

>  A marker interface for {@code Codec} implementations that can derive a new instance that overrides the  {@code UuidRepresentation}.

Reply:
Done